actors/market/tests/deal_termination.rs

```diff
@@ -0,0 +1,221 @@
+use fil_actor_market::{DealSettlementSummary, EX_DEAL_EXPIRED};
+use fil_actors_runtime::EPOCHS_IN_DAY;
+use fvm_shared::{clock::ChainEpoch, econ::TokenAmount, error::ExitCode};
+
+mod harness;
+use harness::*;
+use num_traits::Zero;
+
+#[test]
+fn deal_is_terminated() {
+    struct Case {
+        name: &'static str,
+        deal_start: ChainEpoch,
+        deal_end: ChainEpoch,
+        activation_epoch: ChainEpoch,
+        termination_epoch: ChainEpoch,
+        settlement_epoch: ChainEpoch,
+        payment: TokenAmount,
+    }
+
+    let cases = [
+        Case {
+            name: "deal is terminated after the startepoch and then settle payments before the endepoch",
+            deal_start: 10,
+            deal_end: 10 + 200 * EPOCHS_IN_DAY,
+            activation_epoch: 5,
+            termination_epoch: 15,
+            settlement_epoch: 20,
+            payment: TokenAmount::from_atto(50), // (15 - 10) * 10 as deal storage fee is 10 per epoch
+        },
+        Case {
+            name: "deal is terminated after the startepoch and then settle payments after the endepoch",
+            deal_start: 10,
+            deal_end: 10 + 200 * EPOCHS_IN_DAY,
+            activation_epoch: 5,
+            termination_epoch: 15,
+            settlement_epoch: 10 + 200 * EPOCHS_IN_DAY + 10,
```

Comment:
Idea: you could possibly simplify things here by removing the settlement epoch, and then always attempting settlement (1) shortly after termination, (2) after start epoch, if not already elapsed, and (3) after end epoch.